src/models/__tests__/TeamPlayerSelector.spec.ts

```diff
@@ -7,9 +8,29 @@ import TeamPlayerSelector from "../TeamPlayerSelector";
 import OptimalTeamPlayerSelector from "../TeamPlayerSelector";
 import {SelectorErrorCode } from "../TeamPlayerSelector";
 import GameSkill from "../GameSkill";
+import RandomSource from "../RandomSource";
+import DefaultRandomSource from "../RandomSource";
 
+class OptimalTeamSelectorUnderTest extends OptimalTeamPlayerSelector {
+    constructor(randomSource: RandomSource = new DefaultRandomSource()){
+        super(randomSource);
+    }
+
+    public validateSelectorInput(players: Player[], team: Team): SelectorErrorCode | undefined {
```

Comment:
True but semantically the enum should then be renamed to `ResultCode` rather than `ErrorCode` which is in principle no problem but if you take a look at the interface of the generator it returns either an array of teams or this ErrorCode indicating that the TeamArray means success and the latter happens only on error. Hence, `ErrorCode`. 

Undefined is my current solution to signal success of the subroutine. Not pretty I admit. 

Alternatively we could let the `generate` function of the interface return a tupple `[ResultCode, Array<Team> | undefined]` ?

Reply:
Ah sorry this is the selector but it's the same mechanism. So if we wanna change it for the `TeamPlayerSelector` we should change it for the `TeamGenerator` interface as well.

Reply:
I see, then its fine for me - no necessity for a tuple returnment.